core/services/ocr2/plugins/ocr2vrf/blockhashes/fixed_blockhash_provider.go

```diff
@@ -41,59 +49,77 @@ func (b *fixedBlockhashProvider) OnchainVerifiableBlocks(
 	if err != nil {
 		return 0, nil, errors.Wrap(err, "current height")
 	}
-	fromBlock := toBlock - b.lookbackBlocks
-	var reqs []rpc.BatchElem
 
+	fromBlock := uint64(0)
+	if toBlock > b.lookbackBlocks {
+		fromBlock = toBlock - b.lookbackBlocks
+	}
+
+	var blockHeights []uint64
 	for i := fromBlock; i <= toBlock; i++ {
-		req := rpc.BatchElem{
-			Method: "eth_getBlockByNumber",
-			Args:   []interface{}{hexutil.EncodeBig(big.NewInt(int64(i))), false},
-			Result: &evmtypes.Head{},
+		blockHeights = append(blockHeights, i)
+	}
+
+	blockhashes := make([]common.Hash, len(blockHeights))
+
+	heads, err := b.lp.GetBlocks(blockHeights, pg.WithParentCtx(ctx))
+	if err != nil {
+		b.logger.Warnw("error in get blocks", "err", err)
+	}
+
+	for _, h := range heads {
+		blockhashes[h.BlockNumber-int64(fromBlock)] = h.BlockHash
+	}
+
+	// Fallback to RPC for blocks not found in log poller
```

Comment:
You need this to unblock? Might be a similar amount of work to just add all the blockhashes in the poller

Reply:
yeah I agree it's similar amount of work to add an optional flag to `getBlocks` to fallback to RPC. Perhaps I can just pick that work up now. Wasn't too sure if we wanted to prioritize the benchmarking saving all blocks first and the fallback to RPC functionality might become obsolete.

Reply:
PR for LP: https://github.com/smartcontractkit/chainlink/pull/7419